parlai/core/worlds.py

```diff
@@ -54,12 +54,6 @@
 def validate(observation):
     """Make sure the observation table is valid, or raise an error."""
     if observation is not None and type(observation) == dict:
-        if ('text_candidates' in observation and
```

Comment:
I was thinking I would like to see both my generation and ranking metrics separately once I add ranking. That is, accuracy would be based on generation and ranking based on my candidates. This check in validation forces them to be synced (aka overrides generation if we do ranking), which didn't seem necessary.